src/item.cpp

```diff
@@ -3096,6 +3096,10 @@ double item::get_relative_rot() const
 {
     return goes_bad() ? rot / type->comestible->spoils : 0;
 }
+double item::get_relative_rot( const double make_time ) const
+{
+    return to_turns<double>( rot ) / ( to_turns<double>( type->comestible->spoils ) + make_time );
```

Comment:
Sorry I haven't had a chance to review this earlier.

I was testing this and going over the math, and I'm not sure it adds up.

item.rot is presumably somewhat updated at this point, so that's some rot value greater than what it was when the crafting started.

The relative rot this is retrieving is a number between 0.0 and 1.0 that tells us how far along in the spoilage of the food we are.
rot / type->comestible->spoils was already giving us a value that said that the input ingredient had aged during the craft time.
rot / ( spoils + make_time ) gives us something in between the spoilage when crafting started and the spoilage when it ends, but I'm not sure it's weighted far enough toward the start to be meaningful.

effectively this is like:
( pre-craft rot + craft time ) / ( total rot time + craft_time ) 
If craft time is large compared to total rot time, this is going to pull it closer to 1.0, which is rotten.

I think what you wanted to do is:
( pre-craft rot + craft time - craft time ) / total rot time
i.e.
( rot - craft time ) / total rot time

Reply:
You can calculate that number to subtract with ``get_rot_since( since, until, location )``